packages/query/src/helpers/identify-txs.ts

```diff
@@ -7,16 +7,27 @@ import { SpendableNoteRecord, SwapRecord } from '@penumbra-zone/protobuf/penumbr
 import { Transaction } from '@penumbra-zone/protobuf/penumbra/core/transaction/v1/transaction_pb';
 import { TransactionId } from '@penumbra-zone/protobuf/penumbra/core/txhash/v1/txhash_pb';
 import { sha256Hash } from '@penumbra-zone/crypto-web/sha256';
-import { MsgRecvPacket } from '@penumbra-zone/protobuf/ibc/core/channel/v1/tx_pb';
+import {
+  MsgAcknowledgement,
+  MsgRecvPacket,
+  MsgTimeout,
+} from '@penumbra-zone/protobuf/ibc/core/channel/v1/tx_pb';
 import { FungibleTokenPacketData } from '@penumbra-zone/protobuf/penumbra/core/component/ibc/v1/ibc_pb';
 import { ViewServerInterface } from '@penumbra-zone/types/servers';
 import { parseIntoAddr } from '@penumbra-zone/types/address';
+import { Packet } from '@penumbra-zone/protobuf/ibc/core/channel/v1/channel_pb';
 
 export const BLANK_TX_SOURCE = new CommitmentSource({
   source: { case: 'transaction', value: { id: new Uint8Array() } },
 });
 
-// Identifies if a tx with a relay action of which the receiver is the user
+/**
+ * Identifies if a tx with a relay action of which the receiver is the user.
+ * In terms of minting notes in the shielded pool, three IBC actions are relevant:
+ * - MsgRecvPacket (containing an ICS20 inbound transfer)
+ * - MsgAcknowledgement (containing an error acknowledgement, thus triggering a refund on our end)
+ * - MsgTimeout
```

Comment:
comment: not sure what these other [enum variants](https://github.com/penumbra-zone/penumbra/blob/main/crates/core/component/ibc/src/ibc_action.rs#L42-L55) are?

Reply:
I'd imagine not relevant as those cannot be traced to a specific user and thus in sync we don't need to save anything if we'd encounter it